### PR TITLE
Add missing header files to package

### PR DESCRIPTION
**Pull request by rdspring1**
Fixes issues #817 and #818

Install header files in `nvfuser/include/nvfuser` directory

### CODE REVIEW

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

File: CMakeLists.txt [hunk 1]
```diff
@@ -300,7 +301,19 @@ install(DIRECTORY "${NVFUSER_SRCS_DIR}/"
...
+  PATTERN "csrc/struct.inl")
+
+# TODO guard including flatbuffers headers
+# installing flatbuffers headers
+install(DIRECTORY "${NVFUSER_THIRD_PARTY_DIR}/flatbuffers/include/flatbuffers/"
+  DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/nvfuser/flatbuffers")
+
+# installing dynamic_type headers
+install(DIRECTORY "${NVFUSER_SRCS_DIR}/lib/dynamic_type/src/"
+  DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/nvfuser/"
```
Comment (jjsjann123) [inline CMakeLists.txt:314]: any reason that we don't have a separate dynamic_type folder for the header?

File: setup.py [hunk 2]
```diff
@@ -348,7 +349,8 @@ def main():
             "include/nvfuser/ir/*.h",
             "include/nvfuser/python_frontend/*.h",
             "include/nvfuser/scheduler/*.h",
-            "include/nvfuser/serde*.h",
+            "include/nvfuser/serde/*.h",
+            "include/nvfuser/flatbuffers/*.h",
```
Comment (jjsjann123) [inline setup.py:353]: imagine that once we have dynamic_type header in separate folder, we'll also want to package that?!

**Review by zasdfgbnm — COMMENTED** [review]
LGTM, but will let @xwang233 or @jjsjann123 to approve

**Review by jjsjann123 — COMMENTED** [review]
Minor comments

**Comment by rdspring1** [comment]
!build

File: CMakeLists.txt (hunk 1 above)
Comment (rdspring1) [reply CMakeLists.txt]: I moved the `dynamic_type` headers to a separate folder.

**Review by jjsjann123 — APPROVED** [review]
thanks for cleaning up the dynamic_type thing.

**Comment by jjsjann123** [comment]
Was the codegen diff thing real or just a flaky string/variable name thing?

**Comment by naoyam** [comment]
> Was the codegen diff thing real or just a flaky string/variable name thing?

Looks like they happened with matmul tests. There's a known issue. #799

**Comment by rdspring1** [comment]
This PR should only change what files are included in the nvfuser python package. I'm not sure why it would change the generated code.

**Comment by zasdfgbnm** [comment]
Looks like this PR conflict with some recently merged change, and now nvFuser fail to build. I will patch it at https://github.com/NVIDIA/Fuser/pull/835 then merge.

**Comment by jjsjann123** [comment]
Oops, looks like it's already merged... is `main` broken now? :fearful:

**Comment by zasdfgbnm** [comment]
Yes

**Pull request merged**